### PR TITLE
Asman Umbetov Task 71-80

### PR DESCRIPTION
**Pull request by fastndead**

### CODE REVIEW

File: tasks/75_complementary-dna/complementary-dna.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * @description
+ * takes each character in string and turns it into its' base pair
+ * @example
+ * // returns 'TAACG'
+ * DNAStrand("ATTGC")
+ * @param {String} str - incoming dna material
+ * @return {String}
+ */
+export function DNAStrand(str) {
+  return str.split('').map((item) => {
```
Comment (akimovaa) [inline tasks/75_complementary-dna/complementary-dna.js]: try to use Associative array.

File: tasks/74_dna-pairing/dna-pairing.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * takes each character in string and turns it into pair of that character
+ * and the base pair
+ * @example
+ * // returns [['G', 'C'], ['C','G'],['G', 'C']]
+ * pairElement('GCG')
+ * @param {String} str - incoming dna material
+ * @return {Array} - pairs
+ */
+export function pairElement(str) {
+  return str.split('').map((item) => {
```
Comment (akimovaa) [inline tasks/74_dna-pairing/dna-pairing.js]: try to use Associative array.

File: tasks/76_deep-object-clone/deep-object-clone.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * // returns { a: 1, b: { c:10 } }
+ * deepClone({ a: 1, b: { c:10 } })
+ * @param {Object} obj - object to copy
+ * @return {Object} - copy itself
+ */
+export function deepClone(obj) {
+  const copy = {};
+  for (let key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      if (Array.isArray(obj[key])) {
```
Comment (akimovaa) [inline tasks/76_deep-object-clone/deep-object-clone.js]: Are you sure that you need Array's check?

File: tasks/78_valid-braces/valid-braces.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * @description
+ * validates if parenthesis is balanced
+ * @example
+ * // returns true
+ * validBraces('[]{}()')
+ * @param {String} str
+ * @return {boolean}
+ */
+export function validBraces(str) {
+  const stackForBraces = [];
```
Comment (akimovaa) [inline tasks/78_valid-braces/valid-braces.js]: redundant code

**Review by akimovaa — DISMISSED** [review]

File: tasks/78_valid-braces/valid-braces.js (hunk 4 above)
Comment (fastndead) [reply tasks/78_valid-braces/valid-braces.js]: what do you mean redundant? you mean that the stack is redundant, or the whole logic is off?

File: tasks/76_deep-object-clone/deep-object-clone.js (hunk 3 above)
Comment (fastndead) [reply tasks/76_deep-object-clone/deep-object-clone.js]: well, I really don't need it, yet it works kinda faster with slice (on my machine 9ms with slice and 17ms without). Should I remove array check?

File: tasks/72_carrying/carrying.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * addTogether(2, 3)
+ * @param {Number} a - number to sum
+ * @param {Number} b - number to sum
+ * @return {(Number|Function)}
+ */
+export function addTogether(a, b) {
+  if (typeof a !== 'number') {
+    return;
+  }
+  if (b === undefined) {
```
Comment (vovcyan) [inline tasks/72_carrying/carrying.js]: !b

File: tasks/73_check-for-palindromes/check-for-palindromes.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ * Checks the string if it's a palindrome
+ * @example
+ * // returns true
+ * palindrome('helleh')
+ * @param {String} str - string to check
+ * @return {boolean}
+ */
+export function palindrome(str) {
+  const palindrome = str.replace(/[\W_]+/g, '').toLowerCase().split('');
+  let leftHalf = palindrome.slice(0, palindrome.length / 2);
```
Comment (vovcyan) [inline tasks/73_check-for-palindromes/check-for-palindromes.js]: const

File: tasks/77_expressions-matter/expressions-matter.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+ * // returns 4
+ * expressionMatter(2, 1, 1)
+ * @param {Number} a
+ * @param {Number} b
+ * @param {Number} c
+ * @return {number}
+ */
+export function expressionMatter(a, b, c) {
+  return Math.max((a + b) * c,
+      a*(b+c),
```
Comment (vovcyan) [inline tasks/77_expressions-matter/expressions-matter.js]: Missed spaces in next 3 lines

File: tasks/79_calculate_primoral_of_a_number/calculate_primoral_of_a_number.spec.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
-// import { numPrimorial } from './calculate_primoral_of_a_number';
+import { numPrimorial } from './calculate_primoral_of_a_number';
 
-xdescribe('Calculate Primorial Of a Number', () => {
+describe('Calculate Primorial Of a Number', () => {
```
Comment (vovcyan) [inline tasks/79_calculate_primoral_of_a_number/calculate_primoral_of_a_number.spec.js]: Remove it

File: tasks/80_sum-of-prime-indexes/sum-of-prime-indexes.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+ * checks if number is prime
+ * @example
+ * // returns true
+ * isPrime(5)
+ * @param {Number} number
+ * @return {boolean}
+ */
+function isPrime(number) {
+  let isPrime = number !== 1;
+  for (let j = 2; j< Math.sqrt(number)+1; j++) {
```
Comment (vovcyan) [inline tasks/80_sum-of-prime-indexes/sum-of-prime-indexes.js]: Missed spaces

**Review by vovcyan — CHANGES_REQUESTED** [review]

**Review by alexkond — APPROVED** [review]

**Review by vovcyan — APPROVED** [review]

**Pull request merged**